src/components/AppBar.jsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState, useEffect } from "react";
+import {
+  AppBar,
+  Toolbar,
+  Typography,
+  IconButton,
+  Button,
+  Avatar,
+  Menu,
+  MenuItem,
+  TextField,
+  Grid,
+} from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+import CancelIcon from "@mui/icons-material/Cancel";
+import { useParams, useNavigate } from "react-router-dom";
+import users from "../users";
```

Comment:
This one naturally should be a hook (or context) that will not be effected by the current URL